app/src/main/java/org/mozilla/fenix/settings/deletebrowsingdata/DeleteBrowsingDataFragment.kt

```diff
@@ -239,6 +239,34 @@ class DeleteBrowsingDataFragment : Fragment(R.layout.fragment_delete_browsing_da
             fragmentView.site_permissions_item
         )
     }
+    // If coming from Open Tab -> Settings, pop back to Home
+    // If coming from Home -> Settings, pop back to Settings
+    private fun returnToDeletionOrigin() {
+        if (checkIfBrowserFragmentInBackStack()) findNavController().popBackStack(R.id.homeFragment, false)
+        else findNavController().popBackStack()
+    }
+
+    // For some reason, the only way you can tell if you came from
+    // BrowserFrag -> Settings OR BrowserFrag -> HomeFrag -> Settings is to check whether
+    // there is a browser fragment entry in the back stack. If there is, it means you came from BrowserFrag -> Settings
+    // if there isn't, it means you came from HomeFrag -> Settings
+    private fun checkIfBrowserFragmentInBackStack(): Boolean {
+        val backStackEC = parentFragmentManager.backStackEntryCount
+        val index = backStackEC - 1
+
+        for (i in 0..index) {
+            if (getResIdFromBackstack(parentFragmentManager.getBackStackEntryAt(i).name) == R.id.browserFragment) {
+                return true
+            }
+        }
+
+        return false
+    }
+
+    private fun getResIdFromBackstack(name: String?): Int {
+        val idString = name?.split("-")
```

Comment:
I assume you've tested and found this naming convention to be consistent?

Reply:
Yes and I've added a failsafe in case the convention changes